graveler/committed/tree/tree.go

```diff
@@ -45,7 +45,7 @@ type Repo interface {
 	// RemoveCommonParts accepts the left and right trees of the diff, and finds the common parts which
 	// exist in both trees.
 	// it returns the left and right trees with common parts filtered.
-	RemoveCommonParts(Left, Right Tree) (*Tree, *Tree, error)
+	RemoveCommonParts(left graveler.TreeID, right graveler.TreeID) (*Tree, *Tree, error)
```

Comment:
synced f2f - returned objects are not really Trees, need to change the API to have an accessible read-only way to get Parts, and a way to return a list of Parts for diff.